.github/workflows/collector-slim.yml

```diff
@@ -73,67 +152,27 @@ jobs:
           job-tag: builder
 
       - name: Create Build VMs
-        if: |
-          matrix.arch == 's390x' &&
-          (github.event_name == 'push' || contains(github.event.pull_request.labels.*.name, 'run-multiarch-builds'))
         run: |
           make -C "${{ github.workspace }}/ansible" create-build-vms
 
-      - name: Checks PR, main and release branches
-        run: |
-          if [[ "${{ github.event_name }}" == 'pull_request' ]]; then
-            echo "COLLECTOR_APPEND_CID=true" >> "$GITHUB_ENV"
-            echo "TRACE_SINSP_EVENTS=1" >> "$GITHUB_ENV"
-
-            if [[ "${{ contains(github.event.pull_request.labels.*.name, 'address-sanitizer') }}" == "true" ]]; then
-              echo "ADDRESS_SANITIZER=true" >> "$GITHUB_ENV"
-            fi
-          fi
-
       - name: Create ansible vars
         run: |
-          {
-            echo "---"
-            echo "stackrox_io_username: ${{ secrets.QUAY_STACKROX_IO_RW_USERNAME }}"
-            echo "stackrox_io_password: ${{ secrets.QUAY_STACKROX_IO_RW_PASSWORD }}"
-            echo "rhacs_eng_username: ${{ secrets.QUAY_RHACS_ENG_RW_USERNAME }}"
-            echo "rhacs_eng_password: ${{ secrets.QUAY_RHACS_ENG_RW_PASSWORD }}"
-            echo "collector_git_ref: ${{ github.ref }}"
-            echo "collector_git_sha: ${{ github.sha }}"
-            echo "collector_builder_tag: ${{ env.COLLECTOR_BUILDER_TAG }}"
-            echo "disable_profiling: ${{ matrix.arch != 'amd64' && matrix.arch != 'arm64' }}"
-            echo "rhacs_eng_image: ${{ env.RHACS_ENG_IMAGE }}"
-            echo "collector_image: ${{ inputs.collector-image }}"
-            echo "collector_tag: ${{ inputs.collector-tag }}"
-          } > ${{ github.workspace }}/ansible/secrets.yml
-
-      - name: Authenticate with GCP
-        uses: 'google-github-actions/auth@v2'
-        with:
-          credentials_json: '${{ secrets.GOOGLE_CREDENTIALS_COLLECTOR_SVC_ACCT }}'
-
-      - name: Setup GCP
-        uses: 'google-github-actions/setup-gcloud@v2'
-
-      - name: Build images
-        if: |
-          (github.event_name == 'push' && matrix.arch != 's390x') ||
-          matrix.arch == 'amd64' ||
-          (contains(github.event.pull_request.labels.*.name, 'run-multiarch-builds') && matrix.arch != 's390x')
-        timeout-minutes: 480
-        run: |
-          ansible-playbook \
-            --connection local \
-            -i localhost, \
-            --limit localhost \
-            -e arch='${{ matrix.arch }}' \
-            -e @'${{ github.workspace }}/ansible/secrets.yml' \
-            ansible/ci-build-collector.yml
-
-      - name: Build s390x image
-        if: |
-          (github.event_name == 'push' && matrix.arch == 's390x') ||
-          (contains(github.event.pull_request.labels.*.name, 'run-multiarch-builds') && matrix.arch == 's390x')
+          cat << EOF > ${{ github.workspace }}/ansible/secrets.yml
```

Comment:
Why is this duplicated?

Reply:
I considered using a single job for creating the file and then propagating it via an artifact, problem is that the file has secrets in it and I'd rather not have those be persisted in a file stored as an artifact. 

We might be able to come up with an alternative, the file itself provides 2 types of variables, some are secrets, some others are runtime variables that aren't secrets. We could use 2 separate files, add a secret to GHA that is the YAML contents of the secrets we need, then dump that directly into one file in the build job and use a separate job to create a file with the runtime variables we need and propagate that as an artifact. This all sounds like a whole bunch of effort when all we need is just a single duplicated step though. We could also abstract the step into an action, but not sure how much that is gonna save, since we still need to explicitly pass secrets into the action.

From my point of view, this small duplication is OK for the time being, WDYT?

Reply:
I think this is okay for now. Maybe this can be fixed later.